tests/marqo_test.py

```diff
@@ -221,6 +226,20 @@ def create_test_index(
         return index_name_to_return
 
     def prepare_cloud_index_for_test(self, index_name: str):
+        """
+        Cleans up documents from the specified cloud index and prepares it for testing.
+
+        This method first cleans up any existing documents in the provided 'index_name'.
+        It then ensures that the 'add_documents' method of the Marqo index is correctly mocked
+        to perform it with proper handling by adding _ids of documents
+        to the self.index_to_documents_cleanup_mapping.
```

Comment:
Add why the the _ids are added: `... for later deletion during <RELEVANT FUNCTIONS>`